### ISSUE TITLE
Remove token from hub_utils.tests.common.py and move to secrets

### ISSUE THREAD

**Issue by adrinjalali** [skops-dev/skops#47]
We should remove the token from the repo and move it to secrets, but we should make sure it works on PRs from forks.

**Comment by merveenoyan** [comment]
@BenjaminBossan would you be up for this one?

**Comment by BenjaminBossan** [comment]
I would be up but I have no idea yet how to actually solve the problem.

**Comment by adrinjalali** [comment]
It's possible to have this work in PRs by doing some magic in GH actions, but then running the tests for users would require them to give a token. On the other hand, it's already the case that users would need to pass their token to create the docs. So maybe that's not a big issue?

**Comment by BenjaminBossan** [comment]
> On the other hand, it's already the case that users would need to pass their token to create the docs

They could still use the checked in token for that.

> running the tests for users would require them to give a token

I don't really see a good solution for that. We could mark all tests that require a valid token (which amounts to the network tests I guess) and if users don't provide a token, skip them. If they make changes that don't require those tests, then we're fine, else they have to provide their own token . At the end of the day, all tests will be run on CI.

**Comment by adrinjalali** [comment]
yeah I guess that works.

**Comment by BenjaminBossan** [comment]
Just noticed we also haven the token [here](https://github.com/skops-dev/skops/blob/2c2dd6eddbd88c4774d297c387d1169533246b97/scripts/clean_skops.py#L12). This one could really be replaced by a GH secret, though I guess we would also need to invalidate it since it's already public, and as long as we still use the same token for tests, nothing is gained.

**Comment by merveenoyan** [comment]
@BenjaminBossan AFAIK once you remove it on 🤗Hub it's invalidated. I think I don't have the user access to add tokens (I don't see the settings tab) maybe @adrinjalali can do it (and then we can replace the token with the secret).

**Comment by adrinjalali** [comment]
yeah that one is the same as the one used by tests to create the repos. So changing it doesn't gain us anything.

**Comment by adrinjalali** [comment]
Closing as `hub_utils` is deprecated away.
